drmemtrace invariant "Branch target not immediately after branch" false positives on i-filtered traces

Not sure what other invariant checker runs we have on filtered traces, but I added a new one tool.drcacheoff.rseq-filter and it fails one in about 20 runs on an rule that cannot apply to i-filtered traces:
```
249: Test timeout computed to be: 1500
249: Running |/usr/bin/cmake;-E;remove_directory /home/derek/dr/build/suite/tests/tool.drcacheoff.rseq-filter.linux.rseq.1839280.1334.dir|
249: Running cmd |/home/derek/dr/build/bin64/drrun;-s;90;-quiet;-debug;-killpg;-stderr_mask;0xC;-dumpcore_mask;0;-code_api;-t;drcachesim;-offline;-subdir_prefix;tool.drcacheoff.rseq-filter;-trace_after_instrs;5K;-L0_filter;--;/home/derek/dr/build/suite/tests/bin/linux.rseq|
249: Running postcmd |/home/derek/dr/build/clients/bin64/drcachesim;-indir;/home/derek/dr/build/suite/tests/tool.drcacheoff.rseq-filter.linux.rseq.1839330.6583.dir;-test_mode;-test_mode_name;rseq_app|
249: CMake Error at /home/derek/dr/src/suite/tests/runmulti.cmake:108 (message):
249:   *** postcmd failed (Child aborted): Trace invariant failure in T1839330 at
249:   ref # 766: Branch target not immediately after branch
249: 
249:   ***
249: 
249: Call Stack (most recent call first):
249:   /home/derek/dr/src/suite/tests/runmulti.cmake:120 (process_cmdline)
249: 
249: 
    Test #249: code_api|tool.drcacheoff.rseq-filter ...***Failed    1.60 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =  54.24 sec

The following tests FAILED:
	249 - code_api|tool.drcacheoff.rseq-filter (Failed)
Errors while running CTest
derek@dynamorio:~/dr/build$ ctest --repeat-until-fail 1000 -V -R rseq-filter
```

```
         759         348:     1839330 ifetch       4 byte(s) @ 0x0000ffffaec91d08 b4005d38   cbz    $0x0000ffffaec928ac %x24
         760         348:     1839330 read         4 byte(s) @ 0x0000ffffcfbf5480 by PC 0x0000ffffaec91d0c
         761         348:     1839330 write       16 byte(s) @ 0x0000ffffcfbf5448 by PC 0x0000ffffaec91d4c
         762         348:     1839330 read         8 byte(s) @ 0x0000ffffaecbd8a0 by PC 0x0000ffffaec91d50
         763         348:     1839330 <marker: timestamp 13325884507231868>
         764         348:     1839330 <marker: tid 1839330 on core 28>
         765         348:     1839330 <marker: instruction count 19833>
         766         349:     1839330 ifetch       4 byte(s) @ 0x0000ffffaec91d74 f901c360   str    %x0 -> +0x0380(%x27)[8byte]
```

--- a/clients/drcachesim/tools/invariant_checker.cpp
+++ b/clients/drcachesim/tools/invariant_checker.cpp
@@ -372,15 +372,19 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         // interleaved stream.  Here we look for headers indicating where an interleaved
         // stream *could* switch threads, so we're stricter than necessary.
         if (knob_offline_ && type_is_instr_branch(shard->prev_instr_.instr.type)) {
-            report_if_false(shard,
-                            !shard->saw_timestamp_but_no_instr_ ||
-                                // The invariant is relaxed for a signal.
-                                // prev_xfer_marker_ is cleared on an instr, so if set to
-                                // non-sentinel it means it is immediately prior, in
-                                // between prev_instr_ and memref.
-                                shard->prev_xfer_marker_.marker.marker_type ==
-                                    TRACE_MARKER_TYPE_KERNEL_EVENT,
-                            "Branch target not immediately after branch");
+            report_if_false(
+                shard,
+                !shard->saw_timestamp_but_no_instr_ ||
+                    // The invariant is relaxed for a signal.
+                    // prev_xfer_marker_ is cleared on an instr, so if set to
+                    // non-sentinel it means it is immediately prior, in
+                    // between prev_instr_ and memref.
+                    shard->prev_xfer_marker_.marker.marker_type ==
+                        TRACE_MARKER_TYPE_KERNEL_EVENT ||
+                    // Instruction-filtered are exempted.
+                    TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
+                            shard->file_type_),
+                "Branch target not immediately after branch");
         }
         // Invariant: non-explicit control flow (i.e., kernel-mediated) is indicated
         // by markers.

